radix-common/src/crypto/bls12381/signature.rs

```diff
@@ -42,7 +42,35 @@ impl Bls12381G2Signature {
     }
 
     /// Aggregate multiple signatures into a single one
-    pub fn aggregate(signatures: &[Bls12381G2Signature]) -> Result<Self, ParseBlsSignatureError> {
+    /// This method does validate provided input signatures.
+    pub fn aggregate(signatures: &[Self]) -> Result<Self, ParseBlsSignatureError> {
+        if !signatures.is_empty() {
+            let sig_first = signatures[0].to_native_signature()?;
+
+            sig_first.validate(true)?;
+
+            let mut agg_sig = AggregateSignature::from_signature(&sig_first);
+
+            for sig in signatures.iter().skip(1) {
+                let sig = sig.to_native_signature()?;
+
+                sig.validate(true)?;
+
+                agg_sig.add_signature(&sig, false)?;
+            }
+            let sig = agg_sig.to_signature();
+
+            Ok(Self(sig.to_bytes()))
+        } else {
+            Err(ParseBlsSignatureError::NoSignatureGiven)
+        }
+    }
+
+    /// Aggregate multiple signatures into a single one.
+    /// This method does not validate provided input signatures, it is left
+    /// here for backward compatibility.
+    /// It is recommended to use `aggregate()` method instead.
+    pub fn aggregate_old(signatures: &[Self]) -> Result<Self, ParseBlsSignatureError> {
```

Comment:
Perhaps `aggregate_anemone`, as per our general naming scheme for versioned configuration?

Reply:
Done